src/linked_hash_map.rs

```diff
@@ -0,0 +1,594 @@
+// Copyright 2013 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+
+//! A HashMap wrapper that holds key-value pairs in insertion order.
+//!
+//! # Examples
+//!
+//! ```rust
+//! use collect::LinkedHashMap;
+//!
+//! let mut map: LinkedHashMap<int, int> = LinkedHashMap::new();
```

Comment:
Updated patch without explicit type of LHM, however, I added type suffix like `1i`.